documentation/dev/docker-compose.rst

```diff
@@ -0,0 +1,80 @@
+
+This compose script can also serve as an inspiration for using FlexMeasures in modern cloud environments (like Kubernetes). For instance, you might want to not build the FlexMeasures image from code, but simply pull the image from DockerHub.
+
+.. todo:: This stack runs FlexMeasures, but misses the background worker aspect. For this, we'll add a redis node and one additional FlexMeasures node, which runs a worker as entry point instead (see `issue 418<https://github.com/FlexMeasures/flexmeasures/issues/418>`_).
+
+
+Run the compose stack
+----------------------
+
+Start the stack like this:
+
+.. code-block:: bash
+
+    docker-compose up
+
+You can see log output in the terminal, but ``docker-compose logs`` is also available to you.
+
+Check ``docker ps`` or ``docker-compose ps`` to see if your containers are running:
+
+
+.. code-block:: console
+
+    ± docker ps
+    CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS                    PORTS                    NAMES
+    6105f6d1c91f        flexmeasures_flexmeasures   "bash -c 'flexmeasur…"   45 seconds ago      Up 44 seconds (healthy)   0.0.0.0:5000->5000/tcp   flexmeasures_flexmeasures_1
+    b48e4b9b113b        postgres                    "docker-entrypoint.s…"   44 hours ago        Up 45 seconds             5432/tcp                 flexmeasures_dev-db_1
+
+
+The FlexMeasures container has a health check implemented, which is reflected in this output and you can see which ports are available on your machine to interact.
+
+You can use ``docker-compose logs`` to look at output. ``docker inspect <container>`` and ``docker exec -it <container> bash`` can be quite useful to dive into details. 
+
+.. todo:: We should provide a way to test that this is working, e.g. a list of steps. Document this, but also include that in our tsc/Release list (as a test step to see if Dockerization still works, plus a publish step for the released version).
+
+
+Configuration
+---------------
+
+You can pass in your own configuration (e.g. for MapBox access token, or db URI, see below) like we described above for running a container: put a file ``flexmeasures.cfg`` into a local folder called ``flexmeasures-instance``.
+
+
+Data
+-------
+
+The postgres database is a test database with toy data filled in when the flexmeasures container starts.
+You could also connect it to some other database, by setting a different `SQLALCHEMY_DATABASE_URI` in the config. 
+
+
+Running tests
+---------------
+
+You can run tests in the flexmeasures docker container, using the database service ``test-db`` in the compose file. Per default, we are using the ``dev-db`` database service.
+
+After you've started the compose stack with ``docker-compose up``, run:
+
+.. code-block:: console
+
+    docker exec -it -e SQLALCHEMY_TEST_DATABASE_URI="postgresql://fm-test-db-user:fm-test-db-pass@test-db:5432/fm-test-db" <flexmeasures-container-name> pytest
```

Comment:
`<flexmeasures-container-name>` was `flexmeasures_flexmeasures_1` for me. The two comments I have about that are:
- The tutorial could explicitly include the name at this point, because it is derived from our definition in `docker-compose.yml`.
- The name could be more descriptive, like `flexmeasures_server_1`, similar to `flexmeasures_test-db_1` and `flexmeasures_dev-db_1`, which also get spun up by running `docker-compose up`. This can be done by changing line 21 of our `docker-compose.yml` from `flexmeasures` to `server`.

Reply:
I agree. Thanks.